scope.go

```diff
@@ -283,4 +306,156 @@ func (s *standardScope) fullyQualifiedName(name string) string {
 	return fmt.Sprintf("%s.%s", s.prefix, name)
 }
 
-var globalClock = clock.New()
+// TestScope is a metrics collector that has no reporting, ensuring that
+// all emitted values have a given prefix or set of tags
+type TestScope interface {
+	Scope
+
+	// Snapshot returns a copy of all values since the last report execution,
+	// this is an expensive operation and should only be use for testing purposes
+	Snapshot() Snapshot
+}
+
+// Snapshot is a snapshot of values since last report execution
+type Snapshot interface {
+	// Counters returns a snapshot of all counter summations since last report execution
+	Counters() map[string]CounterSnapshot
+
+	// Gauges returns a snapshot of gauge last values since last report execution
+	Gauges() map[string]GaugeSnapshot
+
+	// Timers returns a snapshot of timer values since last report execution
+	Timers() map[string]TimerSnapshot
+}
+
+// CounterSnapshot is a snapshot of a counter
+type CounterSnapshot interface {
+	// Name returns the name
+	Name() string
+
+	// Tags returns the tags
+	Tags() map[string]string
+
+	// Value returns the value
+	Value() int64
+}
+
+// GaugeSnapshot is a snapshot of a gauge
+type GaugeSnapshot interface {
+	// Name returns the name
+	Name() string
+
+	// Tags returns the tags
+	Tags() map[string]string
+
+	// Value returns the value
+	Value() float64
+}
+
+// TimerSnapshot is a snapshot of a timer
+type TimerSnapshot interface {
+	// Name returns the name
+	Name() string
+
+	// Tags returns the tags
+	Tags() map[string]string
+
+	// Values returns the values
+	Values() []time.Duration
+}
+
+// mergeRightTags merges 2 sets of tags with the tags from tagsRight overriding values from tagsLeft
+func mergeRightTags(tagsLeft, tagsRight map[string]string) map[string]string {
+	if tagsLeft == nil && tagsRight == nil {
```

Comment:
Nit: Any thoughts on checking left and right being nil to avoid creating new map unnecessarily in some cases?